spec/models/reservation_spec.rb

```diff
@@ -494,6 +494,46 @@
     end
   end
 
+  context 'when manually over-extending reservation' do
+    before(:each) do
+      @em = FactoryGirl.create(:equipment_model)
+      @ei = FactoryGirl.create(:equipment_item, equipment_model: @em)
+
+      reservation.update_attributes(equipment_item: @ei,
+                                    equipment_model: @em,
+                                    start_date:
+                                    Time.zone.today + 1.day,
+                                    due_date:
+                                    Time.zone.today + 2.days)
+
+      @r2 = FactoryGirl.build(:valid_reservation,
+                              equipment_item: @ei,
+                              equipment_model: @em,
+                              start_date: Time.zone.today +
+                                          3.days,
+                              due_date: Time.zone.today +
+                                        4.days)
+    end
+
+    it 'should save reservation' do
```

Comment:
I'm not sure if this (and the next) test are needed; we should try to avoid bloating our test suite so since we know that `reservation` already exists and should be valid, there's no reason to check that saving it works (or that FactoryGirl creates the relevant equipment item).